### ISSUE TITLE
Suggestion for the XP tab

### ISSUE THREAD

**Issue by ManicModron** [fmasa/wfrp-master#269]
One of my players mentioned that the XP tab could be easier to use and the rest of my players agreed.

What we would like would be an "Add XP" button that would add XP into the Current XP field and a "Spend XP" button that would move the entered amount from the Current XP field into the Previously Spent XP field.

A field for "Total XP" that added the two fields together would be welcome as well.  This would have the XP tab match the character sheet.

![image](https://github.com/fmasa/wfrp-master/assets/164831256/1ad0c7af-8f4e-4a26-8000-15c46bcf45c4)

- [ ]  A field for Total XP
- [ ] A button to add XP to the Current field
- [ ] A button to subtract XP from the Current field and add the same amount to the Spent field.

Thank you for a very useful app!  We do mainly play from our character sheets, but the app is a great way for me to set up encounters and reference what the players have without asking for their sheets in the middle of the game.

**Comment by ManicModron** [comment]
My apologies if using the Task buttons isn't the appropriate way to itemize suggestions.  I've never used Github before.

**Comment by fmasa** [comment]
I had the idea of the XP log - a way to adjust XP (spend or gain XP) with a date and optional message.
This would allow you to type one number to adjust Current and Spent XP and get a history of changes.

So you would get something like:

| Date | Description | Change | Current XP |
| -- | -- | -- | -- |
| ... | ... | ... | ... | 120 |
| 2023-12-01 | Fulfilled my short-term ambition by winning the Altdorf Annual Pie-Eating Contest | +50 | 170 |
| 2024-01-01 | Moved from Rat Catcher career level to Sewer Jack | -100 | 70 |

This should work for your use case, right? :slightly_smiling_face: 

> My apologies if using the Task buttons isn't the appropriate way to itemize suggestions. I've never used Github before.

No worries, the task list is fine for this :+1: I generally treat issues as the whole thing and split it into other issues if necessary, so no need for special care when formatting the issue description. :slightly_smiling_face:

**Comment by ManicModron** [comment]
Yeah, that sounds like just what we were asking after.  A total XP field would still be awesome, because of Doomed humans and Long Term Ambitions needing to know how much half the gathered XP is, but a full log sounds awesome.

Thank you again for your app as well as being a responsive developer!
